src/core/data/ions/ions.hpp

```diff
@@ -96,16 +95,16 @@ namespace core
 
         NO_DISCARD vecfield_type& velocity() { return bulkVelocity_; }
 
-        NO_DISCARD std::string densityName() const { return "rho"; }
-        NO_DISCARD std::string massDensityName() const { return "massDensity"; }
+        NO_DISCARD std::string static densityName() { return "rho"; }
```

Comment:
should we keep this now the Field is a view and has a name? (same for massDensity)
we should probably let the Field tell their name?

Reply:
this will work if the density is not in a usable state
only using the field will throw if it's not usable, as we have to get the density, which must be usable, to get the name, unless you go through the compile time resource list

Reply:
Now the Field is a view, it is no longer a pointer so it does not need to be usable for its `name()` method to be called, or am I missing something?

Reply:
> it does not need to be usable for its name() method to be called
I don't disagree, but..

Reply:
now these functions can be dropped